lib/jobs/install-os.js

```diff
@@ -258,49 +256,21 @@ function installOsJobFactory(
 
     /**
      * @memberOf InstallOsJob
-     * Update graph progress bases on the input milestone
-     */
-    InstallOsJob.prototype.updateGraphProgress = function(milestone) {
-        var self = this;
-        //sometimes the requested milestone may not be defined by user, so the milestone may not
-        //exist, just safely exit
-        if (!milestone) {
-            return Promise.resolves();
-        }
-
-        return waterline.graphobjects.findOne({instanceId: self.context.graphId})
-        .then(function(graph) {
-            var graphDescription = milestone.description; //Use task description to overwrite graph
-            var progress = GraphProgress.create(graph, graphDescription);
-            progress.updateTaskProgress(self.taskId,
-                    _.pick(milestone, ['value', 'maximum', 'description']), true);
-            return eventsProtocol.publishProgressEvent(graph.instanceId,
-                    progress.getProgressEventData());
-        })
-        .catch(function(error) {
-            //The event publish failure should be swallowed since it is not critical data and it
-            //should never impact the normal process
-            logger.error('Fail to publish graph progress event', {
-                taskId: self.taskId,
-                graphId: self.context.graphId,
-                error: error
-            });
-        });
-    };
-
-    /**
-     * @memberOf InstallOsJob
      */
     InstallOsJob.prototype._run = function() {
         var self = this;
         return Promise.resolve().then(function() {
             return self._convertInstallDisk();
         }).then(function() {
             self._subscribeRequestProfile(function() {
-                return self.updateGraphProgress(self.options.progressMilestones.requestProfile)
-                .then(function() {
-                    return self.profile;
-                });
+                if(_.has(self.options, 'progressMilestones.requestProfile')) {
```

Comment:
This is not enough, if `requestProfile` is set to `null` or something else, the `_.has` will also return `true`.